src/lib/logical_query_plan/abstract_lqp_node.cpp

```diff
@@ -274,17 +274,18 @@ bool AbstractLQPNode::is_column_nullable(const ColumnID column_id) const {
   return left_input()->is_column_nullable(column_id);
 }
 
-bool AbstractLQPNode::has_matching_ucc(const ExpressionUnorderedSet& expressions) const {
+std::optional<UniqueColumnCombination> AbstractLQPNode::get_matching_ucc(
```

Comment:
We never really access the UCCs, right? We only care about their cacheability later, don't we? So maybe return `std::pair<bool, IsCacheable>` directly?